### ISSUE TITLE
Respecting JSONEncoder's keyEncodingStrategy

### ISSUE THREAD

**Issue by jcbriones** [mattpolzin/OpenAPIReflection#5]
**Issue:**
Found an issue where when I use the`OpenAPIEncodedSchemaType.openAPISchema(using encoder: JSONEncoder)` to create a `JSONSchema` with passing an encoder that has `keyEncodingStrategy` set to `convertToSnakeCase`, the JSONSchema's keys wasn't converting the keys to use convertToSnakeCase and returning as what the variable name is set to.

**How to reproduce:**
1. Create a JSONEncoder with the following:
```swift
let encoder = JSONEncoder()
encoder.keyEncodingStrategy = .convertToSnakeCase
```
2. Create a sample model
```swift
struct SampleObject: Codable, Sampleable, OpenAPIEncodedSchemaType {
    let sampleKey: String
    let sampleValue: String
}
```
3. Initialize the model with any values and use the SampleObject's `openAPISchema(_:)` to generate the schema.
4. Verify the keys of the model as `sampleKey` and `sampleValue` instead of `sample_key` and `sample_value`.

**Expected:**
The expected result should be having the keys as `sample_key` and `sample_value`.

**Comment by mattpolzin** [comment]
This is interesting. Fundamentally the JSONEncoder passed in is used as sparingly as possible during reflection, but I can see how you would want (and reasonably expect) the encoder's settings to be respected RE things like key encoding strategy.

This seems like a worthwhile thing to fix, though I am not sure when I will get the time to tackle it.

Thanks for reporting!

**Comment by jcbriones** [comment]
Yeah of course! Wanting to help improve this! It really is useful
